### PR TITLE
[#2611] feat(server): Add server-side REST API support for topic

### PR DESCRIPTION
**Pull request by mchades**
### What changes were proposed in this pull request?

This PR adds RESTful API implementation of messaging catalog topic operations

### Why are the changes needed?

Part of Kafka catalog support work

Fix: #2611 

### Does this PR introduce _any_ user-facing change?

yes, add several topic operations RESTful API

### How was this patch tested?

UTs

### CODE REVIEW

**Comment by jerryshao** [comment]
@mchades can you please fix the conflicts, thanks.

**Comment by mchades** [comment]
conflicts resolved

File: server/src/main/java/com/datastrato/gravitino/server/web/rest/TopicOperations.java [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+            NameIdentifier ident =
+                NameIdentifier.ofTopic(metalake, catalog, schema, request.getName());
+
+            Topic topic =
+                TreeLockUtils.doWithTreeLock(
+                    ident,
+                    LockType.WRITE,
+                    () ->
+                        dispatcher.createTopic(
+                            ident, request.getComment(), null, request.getProperties()));
```
Comment (jerryshao) [inline server/src/main/java/com/datastrato/gravitino/server/web/rest/TopicOperations.java:106]: Can you please add a comment to explain why it is null here?

**Comment by jerryshao** [comment]
Just one minor comment, overall LGTM.

File: server/src/main/java/com/datastrato/gravitino/server/web/rest/TopicOperations.java (hunk 1 above)
Comment (mchades) [reply server/src/main/java/com/datastrato/gravitino/server/web/rest/TopicOperations.java]: added

**Review by jerryshao — APPROVED** [review]

**Pull request merged**